src/collective/cover/browser/compose.py

```diff
@@ -203,44 +90,54 @@ def _move_selected_content(self, origin_tile, target_tile, obj):
             target_dmgr.set(target_data)
             origin_dmgr.delete()
 
+    def setup(self):
+        self.origin_type = self.request.form.get('origin-type')
+        self.origin_id = self.request.form.get('origin-id')
+        self.target_type = self.request.form.get('target-type')
+        self.target_id = self.request.form.get('target-id')
+
     def render(self):
         """Render a tile after populating it with an object."""
-        origin_type = self.request.form.get('origin-type')
-        origin_id = self.request.form.get('origin-id')
-        target_type = self.request.form.get('target-type')
-        target_id = self.request.form.get('target-id')
-        if not all([origin_type, origin_id, target_type, target_id]):
+        if not all([self.origin_type, self.origin_id, self.target_type, self.target_id]):
             raise BadRequest('Invalid parameters')
-        origin_tile = self.context.restrictedTraverse('{0}/{1}'.format(origin_type, origin_id))
-        target_tile = self.context.restrictedTraverse('{0}/{1}'.format(target_type, target_id))
+        origin_tile = self.context.restrictedTraverse(
+            '{0}/{1}'.format(self.origin_type, self.origin_id))
+        target_tile = self.context.restrictedTraverse(
+            '{0}/{1}'.format(self.target_type, self.target_id))
         uuid = self.request.form.get('uuid')
         obj = uuidToObject(uuid)
         if obj is None:
             self._move_all_content(origin_tile, target_tile)
         else:
             self._move_selected_content(origin_tile, target_tile, obj)
         # reinstantiate the tile to update its content on AJAX calls
-        target_tile = self.context.restrictedTraverse('{0}/{1}'.format(target_type, target_id))
+        target_tile = self.context.restrictedTraverse(
+            '{0}/{1}'.format(self.target_type, self.target_id))
```

Comment:
@rodfersou wouldn't be better to do `'{0}/{1}'.format(self.target_type, self.target_id)` only once?